camelus_lib.ml

```diff
@@ -931,6 +930,109 @@ module PrChecks = struct
+              let open OpamProcess.Job.Op in
+              OpamProcess.Job.run @@
+              OpamFilename.with_tmp_dir_job @@ fun dir ->
+              OpamProcess.Job.catch (function
+                    Failure msg -> Done (`err msg)
+                  | e -> Done (`err (Printexc.to_string e)))
+              @@ fun () ->
+              OpamDownload.download ~overwrite:false (OpamFile.URL.url url) dir
+              @@| fun f ->
+              (match OpamFile.URL.checksum url with
+               | [] -> `warn
+               | chks ->
+                 let not_corresponding =
+                   List.filter (fun chk ->
+                       not( OpamHash.check_file (OpamFilename.to_string f) chk))
+                     chks
+                 in
+                 if not_corresponding = [] then
+                   `succ
+                 else
+                   let msg =
+                     Printf.sprintf "Cheksum%s %s don't verify archive"
+                       (if List.length chks = 1 then "" else "s")
+                       (OpamStd.List.to_string OpamHash.to_string not_corresponding)
+                   in
+                   `err msg
+              )
+            | None -> `fatal
+          in
+          let nv =
+            match OpamPackage.of_filename filename with
+            | Some nv -> nv
+            | None -> OpamPackage.of_string "invalid-package-name.v"
+          in
+          Lwt.return (nv, result)
+        ) (opam_files files)
+    in
+    let ok =
+      OpamStd.List.filter_map (fun (nv,res) ->
+          match res with
+          | `succ -> Some nv
+          | _ -> None) upstream_check
+    in
+    let warn =
+      OpamStd.List.filter_map (fun (nv,res) ->
+          match res with
+          | `warn -> Some nv
+          | _ -> None) upstream_check
+    in
+    let err =
+      OpamStd.List.filter_map (fun (nv,res) ->
+          match res with
+            `err m -> Some (nv, m)
+          | _ -> None) upstream_check
+    in
+    let fatal =
+      OpamStd.List.filter_map (fun (nv,res) ->
+          match res with
+          | `fatal -> Some nv
+          | _ -> None) upstream_check
+    in
+    let title, status =
+      if err <> [] || fatal <> []then
+        "##### :x: upstream errors",
+        `Errors (List.map OpamPackage.to_string (fatal @ (List.map fst err)))
+      else if warn <> [] then
+        "##### :link: opam-lint warnings",
+        `Warnings (List.map OpamPackage.to_string warn)
+      else if ok <> [] then
+        "##### :heavy_check_mark: Upstream checked", `Passed
+      else
+        "##### No new or changed opam files", `Passed
+    in
+    let title =
+      Printf.sprintf "%s <small>%s</small>\n\n" title head
+    in
+    let str_pkg =
+      OpamStd.List.concat_map ~last_sep:", and" ", " OpamPackage.to_string
+    in
+    let warn = Printf.sprintf "No checksum defined for %s" (str_pkg warn) in
+    let err =
+      OpamStd.Format.itemize ~bullet:"    * "
```

Comment:
I am not sure this formatting will work, `itemize` is suposed to start at the beginning of a line.
Maybe just add a header? (e.g. `"Errors:\n" ^ OpamStd.Format.itemize ...`)